packages/terraform-modules-public/testnet/modules/validator/startup.sh

```diff
@@ -0,0 +1,195 @@
+#!/bin/bash
+
+# ---- Set Up Logging ----
+
+curl -sSO https://dl.google.com/cloudagents/install-logging-agent.sh
+bash install-logging-agent.sh
+
+# ---- Set Up Persistent Disk ----
+
+# gives a path similar to `/dev/sdb`
+DISK_PATH=`readlink -f /dev/disk/by-id/google-${attached_disk_name}`
+DATA_DIR=/root/.celo
+
+echo "Setting up persistent disk ${attached_disk_name} at $DISK_PATH..."
+
+DISK_FORMAT=ext4
+CURRENT_DISK_FORMAT=`lsblk -i -n -o fstype $DISK_PATH`
+
+echo "Checking if disk $DISK_PATH format $CURRENT_DISK_FORMAT matches desired $DISK_FORMAT..."
+
+# If the disk has already been formatted previously (this will happen
+# if this instance has been recreated with the same disk), we skip formatting
+if [[ $CURRENT_DISK_FORMAT == $DISK_FORMAT ]]; then
+  echo "Disk $DISK_PATH is correctly formatted as $DISK_FORMAT"
+else
+  echo "Disk $DISK_PATH is not formatted correctly, formatting as $DISK_FORMAT..."
+  mkfs.ext4 -m 0 -F -E lazy_itable_init=0,lazy_journal_init=0,discard $DISK_PATH
+fi
+
+echo "Mounting $DISK_PATH onto $DATA_DIR"
+mount -o discard,defaults $DISK_PATH $DATA_DIR
+[[ ${reset_geth_data} == "true" ]] && rm -rf $DATA_DIR/geth
+mkdir -p $DATA_DIR/account
+
+# ---- Install Docker ----
+
+echo "Installing Docker..."
+apt update && apt upgrade
+apt install -y apt-transport-https ca-certificates curl software-properties-common gnupg2
+curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable"
+apt update && apt upgrade
+apt install -y docker-ce
+systemctl start docker
+
+echo "Configuring Docker..."
+cat <<'EOF' > '/etc/docker/daemon.json'
+{
+  "log-driver": "json-file",
+  "log-opts": {
+    "max-size": "10m",
+    "max-file": "3" 
+  }
+}
+EOF
+systemctl restart docker
+
+# ---- Set Up and Run Geth ----
+
+GETH_NODE_DOCKER_IMAGE=${geth_node_docker_image_repository}:${geth_node_docker_image_tag}
+
+ACCOUNT_ADDRESS=${validator_account_address}
+echo "Address: $ACCOUNT_ADDRESS"
+echo "Private Key: ${validator_private_key}"
+
+
+echo "Proxy enode address: ${proxy_enode}"
+echo "Proxy internal ip address: ${proxy_internal_ip}"
+echo "Proxy external ip address: ${proxy_external_ip}"
+PROXY_INTERNAL_ENODE="enode://${proxy_enode}@${proxy_internal_ip}:30503"
+PROXY_EXTERNAL_ENODE="enode://${proxy_enode}@${proxy_external_ip}:30503"
+
+PROXY_URL="$PROXY_INTERNAL_ENODE;$PROXY_EXTERNAL_ENODE"
+echo "Proxy URL: $PROXY_URL"
+
+echo "Pulling geth..."
+docker pull $GETH_NODE_DOCKER_IMAGE
+
+IN_MEMORY_DISCOVERY_TABLE_FLAG=""
+[[ ${in_memory_discovery_table} == "true" ]] && IN_MEMORY_DISCOVERY_TABLE_FLAG="--use-in-memory-discovery-table"
+
+# Load configuration to files
+echo -n '${genesis_content_base64}' | base64 -d > $DATA_DIR/genesis.json
+echo -n '${static_nodes_base64}' | base64 -d > $DATA_DIR/static-nodes.json
```

Comment:
Should delete this

Separately, maybe letting a validator use a router isn't the right option.. @reem ran this without an external IP address and found himself still peering with the static nodes when that shouldn't have been possible

Reply:
Deleted.
I guess that submitting the `static-nodes.json` the proxy would connect to those nodes but it would not accept any incoming connection from any other node, but also did not start peering new connections as `--no-discover` is configured. 
Now only connection the connection to the proxy should occur.